dev-client/src/components/menus/MenuItem.tsx

```diff
@@ -0,0 +1,158 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useMemo} from 'react';
+import {PressableProps, StyleSheet, View} from 'react-native';
+import {Text, TouchableRipple} from 'react-native-paper';
+
+import {Icon, IconName} from 'terraso-mobile-client/components/icons/Icon';
+import {convertColorProp} from 'terraso-mobile-client/components/util/nativeBaseAdapters';
+
+export type MenuItemVariant = 'default' | 'destructive';
+
+export type MenuItemProps = {
+  variant?: MenuItemVariant;
+  uppercase?: boolean;
+  label: string;
+  subLabel?: string;
+  icon?: IconName | React.ReactElement;
+  pill?: React.ReactNode;
+  selected?: boolean;
+  disabled?: boolean;
+  onPress?: PressableProps['onPress'];
+};
+
+export function MenuItem({
+  variant = 'default',
+  uppercase = false,
+  icon,
+  label,
+  subLabel,
+  pill,
+  selected,
+  disabled,
+  onPress,
+  children,
+}: React.PropsWithChildren<MenuItemProps>) {
+  const variantStyle = useMemo(() => {
+    if (disabled) {
+      return styles.disabled;
+    } else if (variant === 'destructive') {
+      return styles.destructive;
+    } else {
+      return styles.default;
+    }
+  }, [disabled, variant]);
+
+  return (
+    <TouchableRipple
+      onPress={onPress!}
+      disabled={disabled}
+      style={selected ? styles.selected : undefined}
+      accessibilityRole="menuitem"
+      accessibilityState={{selected: selected, disabled: disabled}}
+      accessibilityLabel={`${label} ${subLabel}`}>
+      <View style={styles.root}>
+        <View style={[styles.section, styles.iconSection]}>
+          {typeof icon === 'string' ? (
+            <Icon
+              name={icon as IconName}
+              size="md"
+              color={variantStyle.color}
+            />
+          ) : (
+            (icon ?? <View style={styles.emptyIcon} />)
+          )}
+        </View>
+        <View style={[styles.section, styles.labelSection]}>
+          <View style={[styles.labelsContainer]}>
+            <Text
+              selectable={false}
+              style={[
+                styles.label,
+                uppercase && styles.uppercaseLabel,
+                variantStyle,
+              ]}>
+              {label}
+            </Text>
+            {subLabel && (
+              <Text selectable={false} style={[styles.subLabel]}>
+                {subLabel}
+              </Text>
+            )}
+            {children && <View style={styles.childContainer}>{children}</View>}
+          </View>
+        </View>
+        {pill && (
+          <View style={[styles.section, styles.pillSection]}>{pill}</View>
+        )}
+      </View>
+    </TouchableRipple>
+  );
+}
+
+const styles = StyleSheet.create({
+  root: {
+    flexDirection: 'row',
+    margin: 16,
```

Comment:
Side question: Do you envision moving away from the standardized sizing in theme.ts, and do them in this one-off per-component way? I feel like it's nice to have some set of standards but don't have takes on how (if theme.ts is a nativebase concept)
Was thinking about this because noticing this is the 'space.md' size (which is what we use in the ScreenContentSection, which has the same flavor of "just keep content 16px from the edge of the screen")

Reply:
I think we could eventually have constants in place for these concepts (like we're still pulling in `theme.ts` for colors) but for now I can't think of a better solution. I think it can be something to think about as we continue implementing component stuff.

Reply:
React Native Paper also supports theming (but differently than NativeBase). We have a little bit of RNP theming in `theme.ts` as well.

question: would it make sense to use or extend the ReactNativePaper `Menu.Item` here?

Reply:
On the whole, my opinion is that a unified strategy for theming post-NativeBase is a large-scale task of its own and is out-of-scope here. But we should keep it in mind as we implement more components; might be a later-stage task as our component library matures.